honeybee_grasshopper_energy/src/HB Custom Window Gap Material.py

```diff
@@ -0,0 +1,76 @@
+This material can be plugged into the "HB Window Construction" component.
+_
+This object allows you to specify specific values for conductivity,
+viscosity and specific heat through the following formula:
+    property = A + (B * T)
+where:
+    A, B = regression coefficients for the gas
+    T = temperature [K]
+-
+
+    Args:
+        _name: Text string for material name.
+        _thickness: Number for the thickness of the gas gap layer [m].
+            Default: 0.0125
+        _conductivity_a: First conductivity coefficient.
+            Or condictivity in [W/m-K] if b coefficient is 0.
+        _viscosity_a: First viscosity coefficient.
+            Or viscosity in [kg/m-s] if b coefficient is 0.
+        _specific_heat_a: First specific heat coefficient.
+            Or the specific heat in [J/kg-K] if b coefficient is 0.
+        _conductivity_b_: Second conductivity coefficient. Default = 0.
+        _viscosity_b_: Second viscosity coefficient. Default = 0.
+        _specific_heat_b_: Second specific heat coefficient. Default = 0.
+        _spec_heat_ratio_: A number for the the ratio of the specific heat at
+            contant pressure, to the specific heat at constant volume.
+            Default is 1.0 for Air.
+        _mol_weight_: Number between 20 and 200 for the mass of 1 mol of
+            the substance in grams. Default is 20.0.
+    
+    Returns:
+        mat: A custom gas gap material that describes a layer in a window construction
+            and can be assigned to a Honeybee Window construction.
+"""
+
+ghenv.Component.Name = "HB Custom Window Gap Material"
+ghenv.Component.NickName = 'CustomGapMat'
+ghenv.Component.Message = '0.1.0'
+ghenv.Component.Category = "Energy"
+ghenv.Component.SubCategory = "1 :: Construction"
+ghenv.Component.AdditionalHelpFromDocStrings = "5"
+
+
+try:  # import the honeybee-energy dependencies
+    from honeybee_energy.material.gas import EnergyWindowMaterialGasCustom
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_energy:\n\t{}'.format(e))
+try:  # import ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+
+if all_required_inputs(ghenv.Component):
+    # set the default material properties
+    _conductivity_b_ = 0 if _conductivity_b_ is None else _conductivity_b_
+    _viscosity_b_ = 0 if _viscosity_b_ is None else _viscosity_b_
+    _specific_heat_b_ = 0 if _specific_heat_b_ is None else _specific_heat_b_
+    _spec_heat_ratio_ = 1.0 if _spec_heat_ratio_ is None else _spec_heat_ratio_
+    _mol_weight_ = 20.0 if _mol_weight_ is None else _mol_weight_
+    
+    # create the material
+    mat = EnergyWindowMaterialGasCustom(
+        _name, _thickness, _conductivity_a, _viscosity_a, _specific_heat_a,
+        _conductivity_b_, _viscosity_b_, _specific_heat_b_, 0, 0, 0,
```

Comment:
what are these 0, 0, 0 values. I will be more readable and easier to maintain in the future if we define them as variables and pass the variables to the class.

Reply:
They are `_conductivity_c_, _viscosity_c_, _specific_heat_c_`. So it was kinda infer-able from the pattern there but I'll make it more explicit anyway.